docs/readmes/resources/ref_magma_metrics.md

```diff
@@ -0,0 +1,186 @@
+[magmad.yml](https://sourcegraph.com/github.com/magma/magma/-/blob/feg/gateway/configs/magmad.yml).
+Then, `magmad`schedules its [`MetricsCollector`]() object to collect and upload
+metrics every `metrics_config.sync_interval` seconds. The `MetricsCollector`
+object loops over all Python services and for each, calls `GetMetrics` over gRPC
+to obtain metrics from the Python service. Then, it divides the gRPC structures
+into chunks of 1 MB or less, and uploads these chunks to the `metricsd` orc8r
+service over gRPC through the `Collect` method in `metricsd`.
+
+```python
+chan = ServiceRegistry.get_rpc_channel(
+    'metricsd',
+    ServiceRegistry.CLOUD,
+    grpc_options=self._grpc_options,
+)
+client = MetricsControllerStub(chan)
+...
+sample_chunks = self._chunk_samples(samples)
+for idx, chunk in enumerate(sample_chunks):
+    ...
+    future = client.Collect.future(
+        metrics_container,
+        self.grpc_timeout,
+    )
+```
+
+In **C++** land, each service also exposes the `GetMetrics` method that the
+`magmad` service uses to fetch collected metrics over gRPC. The
+[`MetricsSingleton`](https://sourcegraph.com/github.com/magma/magma/-/blob/orc8r/gateway/c/common/service303/MetricsSingleton.cpp)
+class provides helpers and wrappers around the Prometheus C++ client.
+[`service303.cpp`](https://sourcegraph.com/github.com/magma/magma/-/blob/lte/gateway/c/core/oai/tasks/service303/service303.cpp)
+further wraps `MetricsSingleton` methods to make it even more convenient for
+Magma services to set and upload Prometheus metrics.
+
+### Phase 2: Orchestrator and Prometheus Edge Hub
+
+The `metricsd` cloud service reads metric names and labels from enums defined in
+[`metricsd.proto`](https://sourcegraph.com/github.com/magma/magma/-/blob/orc8r/protos/metricsd.proto)
+and accepts gRPC messages accordingly. Once it has received metrics from the
+AGW, it pushes them again over gRPC to the
+[`MetricsExporter`](https://sourcegraph.com/github.com/magma/magma/-/blob/orc8r/cloud/go/services/metricsd/protos/exporter.proto)
+servicers in its `Collect()` method.
+
+```go
+metricsExporters, err := metricsd.GetMetricsExporters()
+...
+for _, e := range metricsExporters {
+	err := e.Submit(metricsToSubmit)
+    ...
+}
+```
+
+With default configuration, `GRPCPushExporterServicer` would be the servicer
+registered for `MetricsExporter` which is registered in the
+[`orchestrator`](https://sourcegraph.com/github.com/magma/magma/-/blob/orc8r/cloud/go/services/orchestrator/orchestrator/main.go?L66)
+cloud service. Finally, the `GRPCPushExporterServicer`, through its `Submit`
+method, pushes metrics over gRPC to their home on Prometheus Edge Hub.
+
+```go
+client, err := s.getClient()
+if err != nil {
+    return err
+}
+_, err = client.Collect(context.Background(), &edge_hub.MetricFamilies{Families: families})
+```
+
+The following diagram shows the call flow for the metrics pipeline on the orc8r:
+
+![Orc8r Metrics](assets/orc8r/orc8r_metrics.png)
```

Comment:
Bring this much closer to the top of the document, if not in the intro. "BLUF" is a good way to go, then fill in the details with words

https://www.internalfb.com/intern/wut/word/?word=bluf